influxdb_store.go

```diff
@@ -647,8 +646,15 @@ func addChildren(root *Trace, children []*Trace) error {
 		if len(children) == 0 {
 			break
 		}
-		if try == retries {
-			return errMaxRetries
+		if try == retries { // At this point all children were added to their parent.
+
+			// Iterates over children(without parent found on `root`) and appends them as sub-traces to `root`.
+			for _, child := range children {
```

Comment:
To ensure this code is easily readable in the future, we should try to write a more clear comment here. How about something like this?

```
// At this point, all children were added to their parent spans. Any children
// left over in the children slice do not have parents. This could happen if,
// for example, a parent service fails to record its span information to the
// collection server but its downstream services do send their span information
// properly. In this case, we gracefully degrade by adding these orphan spans to
// the root span.
```